### PR TITLE
Upgrades generated TypeScript bindings to the latest soroban-client 

### PR DESCRIPTION
**Pull request by Shaptic**
### What
This upgrades `soroban-client` to [v1.0.0-beta.2](https://github.com/stellar/js-soroban-client/releases/tag/v1.0.0-beta.2) and migrates the project template code accordingly. The changes can be summarized as:

 * leveraging the new `string -> ScVal { ScAddress }` capability of `nativeToScVal`
 * dealing with the new `SimulateTransactionResponse` schema
 * leveraging the new `GetTransactionStatus` enum

### Why
Self-explanatory: staying up to date with the pending testnet release.

### CODE REVIEW

File: cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ This library was automatically generated by Soroban CLI using a command similar
 
 ```bash
 soroban contract bindings ts \
-  --rpc-url https://rpc-futurenet.stellar.org:443 \
+  --rpc-url https://rpc-futurenet.stellar.org:443/soroban/rpc \
```
Comment (tsachiherman) [inline cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md:9]: was this intentional ?

**Review by tsachiherman — COMMENTED** [review]
just one.

File: cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md (hunk 1 above)
Comment (Shaptic) [reply cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md:9]: hm, nope. not sure where that came from, actually. looking :detective:

Comment (Shaptic) [reply cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md:9]: nice catch - looks like the boilerplate wasn't updated

Comment (tsachiherman) [reply cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/README.md:9]: now I'm even more confused : you updated the boilerplate, but left the opposite in here ?!

File: cmd/crates/soroban-spec-typescript/src/boilerplate.rs [hunk 2]
```diff
@@ -145,7 +145,7 @@ mod test {
         p.init(
             "test_custom_types",
             "CA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQGAXE",
-            "https://rpc-futurenet.stellar.org:443/soroban/rpc",
+            "https://rpc-futurenet.stellar.org:443",
```
Comment (sreuland) [inline cmd/crates/soroban-spec-typescript/src/boilerplate.rs:148]: good catch, that url would never have worked, I think most using these bindings were overriding the gen with `--network` to get around this.

**Review by sreuland — APPROVED** [review]
thanks for resolving this, [soroban-examples-dapp uses the ts bindings](https://github.com/stellar/soroban-example-dapp/blob/main/package.json#L13), so this will help greatly in bringing that up to latest release as well.

**Review by tsachiherman — APPROVED** [review]

File: cmd/crates/soroban-spec-typescript/src/project_template/src/invoke.ts [hunk 3]
```diff
@@ -102,38 +103,30 @@ export async function invoke<R extends ResponseTypes, T = string>({
     fee: fee.toString(10),
     networkPassphrase,
   })
+    .setNetworkPassphrase(networkPassphrase)
```
Comment (willemneal) [inline cmd/crates/soroban-spec-typescript/src/project_template/src/invoke.ts:106]: Isn't this added two lines above when initializing the builder?
```suggestion
```

Comment (Shaptic) [reply cmd/crates/soroban-spec-typescript/src/project_template/src/invoke.ts:106]: uhh yeah, dunno why that happened

**Pull request merged**